### PR TITLE
secondLargestNum implemented with tests

### PR DESCRIPTION
**Pull request by linhapeter**

### CODE REVIEW

File: src/tasks/secondLargestNum.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const secondLargestNum = (arr) => {
+  if (arr.length < 2) {
+    throw new Error("Array needs at least 2 elements");
+  }
+  const arrWithoutLargest = arr.filter((number) => number != Math.max(...arr));
+  const result = Math.max(...arrWithoutLargest);
+  return result;
```
Comment (igorTT) [inline src/tasks/secondLargestNum.js]: Please check the complexity of this function and reduce it to 0(n). Currently I see 0(n ** 2)

File: src/tests/secondLargestNum.test.js [hunk 2]
```diff
@@ -5,8 +5,9 @@ describe("secondLargestNum", () => {
     expect(secondLargestNum([1, 2, 3, 4, 5])).toEqual(4);
     expect(secondLargestNum([5, 1, 3, 7, 2])).toEqual(5);
     expect(secondLargestNum([10, 20, 30, 40, 50])).toEqual(40);
-    expect(secondLargestNum([5])).toThrow("Array needs at least 2 elements");
-    expect(secondLargestNum([])).toThrow("Array needs at least 2 elements");
+    // expect(secondLargestNum([5])).toThrow("Array needs at least 2 elements");
+    // expect(secondLargestNum([])).toThrow("Array needs at least 2 elements");
```
Comment (igorTT) [inline src/tests/secondLargestNum.test.js]: Useless comments

File: src/tasks/secondLargestNum.js [hunk 3]
```diff
@@ -2,7 +2,15 @@ export const secondLargestNum = (arr) => {
...
-  return result;
+
+  let first = Number.NEGATIVE_INFINITY;
+  let second = Number.NEGATIVE_INFINITY;
+
+  for (let index = 0; index < arr.length; index++) {
+    if (arr[index] > first) {
+      second = first;
+      first = arr[index];
+    } else if (arr[index] > second) second = arr[index];
```
Comment (igorTT) [inline src/tasks/secondLargestNum.js]: Please wrap to the curly braces. Think about readability first.

File: src/tasks/secondLargestNum.js [hunk 4]
```diff
@@ -2,7 +2,15 @@ export const secondLargestNum = (arr) => {
   if (arr.length < 2) {
     throw new Error("Array needs at least 2 elements");
   }
-  const arrWithoutLargest = arr.filter((number) => number != Math.max(...arr));
-  const result = Math.max(...arrWithoutLargest);
-  return result;
+
+  let first = Number.NEGATIVE_INFINITY;
+  let second = Number.NEGATIVE_INFINITY;
```
Comment (igorTT) [inline src/tasks/secondLargestNum.js]: What reasons for adding -Infinity? Don't overcomplicate. Declare first and second with empty values.

Comment (linhapeter) [reply src/tasks/secondLargestNum.js]: Unfortunately it doesn't work with empty values.

Comment (igorTT) [reply src/tasks/secondLargestNum.js]: Try this one
image

**Review by igorTT — APPROVED** [review]

**Pull request merged**